### PR TITLE
Fix failed free trial checkout when card requires authentication

### PR DESCRIPTION
**Pull request by dechov**
Fixes https://github.com/Automattic/woocommerce-payments/issues/2325

#### Changes proposed in this Pull Request

When the order does not require payment (such as a free trial-only order), a setup intent is created instead of a payment intent. Since the former is not deserialized into an `WC_Payments_API_Intention` object, it needs to be treated differently for the purpose of reading the "next action" from it.

Adds an automated test covering this case.

#### Testing instructions

- Put only subscription product(s) with free trial into cart
- Attempt to check out with 4000002500003155

-------------------

- [ ] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

### CODE REVIEW

File: tests/unit/test-class-wc-payment-gateway-wcpay-process-payment.php [hunk 1]
```diff
@@ -703,6 +703,124 @@ function( $note ) use ( $intent_id, $total ) {
 		);
 	}
 
+	/**
+	 * Test processing free order with the status "requires_action".
+	 * This is the status returned when the saved card setup requires
+	 * further authentication with 3DS.
+	 */
+	public function test_setup_intent_status_requires_action() {
```
Comment (dechov) [inline tests/unit/test-class-wc-payment-gateway-wcpay-process-payment.php:711]: Mostly copied from previous test, the main difference being $0 total.

**Review by diegocurbelo — COMMENTED** [review]
The checkout works correctly now, but issuing a manual renewal fails:

Screen_Shot_2021-07-09_at_19_18_49

**Comment by dechov** [comment]
@diegocurbelo That will be fixed with @DanReyLop's change [here](https://github.com/Automattic/woocommerce-payments/pull/2369/files#diff-bcfd303d507c33a27671935e724ee64d0df30ff2a18bb6ef4c94f28d48190124R1918)

**Review by diegocurbelo — APPROVED** [review]
The checkout flow works well, and the subscription bug I commented is being will be fixed in #2369... we can merge this 🚢

**Pull request merged**